### PR TITLE
Text events, plugin events rewrite 

### PR DESCRIPTION
**Pull request by kramitox**
Rewrote how events are handled to work properly and conform to eslint.

Added:
Text events - Follow / Sub / Joined / Spell

### CODE REVIEW

**Comment by Takkes** [comment]
Please remove all the latest-follow.txt etc files. Those shouldn't be in source control. Also add them to the gitignore then.

File: plugins/alert-spell/alert-spell.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  permissions: [], // This is for Permissisons depending on the Platform.
+  alias: [], // Alias commands that preform interesting things.
+  cooldown: 0, // this is Set in Seconds, how long between the next usage of this command.
+  settings: false, // Defining this as false will load the Settings file for this Plugin when the system loads this plugin.
+  credits: `New Spell system by Krammy. Original by Bioblaze Payne.`, // MAKE SURE YOU FILL THIS IN GOD DAMNIT!
+  execute(client, data) {
+    client.sendMessage(`Thanks @${data.user} for the casting a spell <3`);
+
+    if (settings.output)
+      fs.writeFile('./plugins/alert-spell/latest-spell.txt', data.user, (err) => {
```
Comment (Bioblaze) [inline plugins/alert-spell/alert-spell.js:21]: couldn't we save the Spell they are casting, as well as the # they cast, and if it was Elixir or Mana?

File: plugins/alert-sub/alert-sub.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  permissions: [], // This is for Permissisons depending on the Platform.
+  alias: [], // Alias commands that preform interesting things.
+  cooldown: 0, // this is Set in Seconds, how long between the next usage of this command.
+  settings: false, // Defining this as false will load the Settings file for this Plugin when the system loads this plugin.
+  credits: `New Subscriber system by Krammy. Original by Bioblaze Payne.`, // MAKE SURE YOU FILL THIS IN GOD DAMNIT!
+  execute(client, data) {
+    client.sendMessage(`Thanks @${data.user} for the Subscription <3`);
+
+    if (settings.output)
+      fs.writeFile('./plugins/alert-sub/latest-sub.txt', data.user, (err) => {
```
Comment (Bioblaze) [inline plugins/alert-sub/alert-sub.js:21]: we should record the time here in a `csv` basically just  array seperated by `,` thats all. so when we read line by line, we can parse the line based on `,` then get the value.

File: plugins/drpc-viewercount-update/drpc-viewercount-update.js [hunk 3]
```diff
@@ -15,13 +15,14 @@ module.exports = {
   execute(client, data) {
     const discord = Modules.getModule('discord-rpc');
     if (discord) {
-      if (discord.settings.active) {
-        discord.setCount(Number(data['live.viewers']));
-      }
+      if (discord.settings !== undefined)
```
Comment (Bioblaze) [inline plugins/drpc-viewercount-update/drpc-viewercount-update.js:18]: this should never be undefined, <.< this should check if it is null. not undefined.

File: plugins/alert-sub/alert-sub.js (hunk 2 above)
Comment (kramitox) [reply plugins/alert-sub/alert-sub.js:21]: No as OBS cannot parse this itself when being loaded straight from text file. See Discord #developers for example.
We can do this in a seperate file if you wish.

File: plugins/drpc-viewercount-update/drpc-viewercount-update.js (hunk 3 above)
Comment (kramitox) [reply plugins/drpc-viewercount-update/drpc-viewercount-update.js:18]: This gave underfined in the output. Error in your code? Make sure you're not just using your custom build but also pulling a fresh clone of the repo to test your stuff.

File: plugins/alert-spell/alert-spell.js (hunk 1 above)
Comment (kramitox) [reply plugins/alert-spell/alert-spell.js:21]: In a different file yes. As this file is being used just to save who casted it for on-screen showing in OBS and Slobs no.

**Pull request closed without merging**